### PR TITLE
feat: add more tests for proxy.ts

### PR DESCRIPTION
**Pull request by jsjoeio**
This PR adds more tests for `proxy.ts`, specifically to test that our proxy handles error events appropriately.

Fixes N/A

P.S. huge shoutout to @vapurrmaid for pairing with me on this 🎉

### CODE REVIEW

**Comment by jsjoeio** [comment]
cc @code-asher would love to get your eyes on this and help me understand what I'm missing

File: test/unit/proxy.test.ts [hunk 1]
```diff
@@ -102,4 +108,135 @@ describe("proxy", () => {
...
+
+  afterEach(async () => {
+    await testServer.close()
+    await proxyTarget.close()
+  })
+
+  it("should return a 500 when proxy target errors ", async () => {
+    // Close the proxy target so that proxy errors
+    await proxyTarget.close()
+    const errorResp = await nodeFetch.default(`${URL}/error`)
```
Comment (vapurrmaid) [inline test/unit/proxy.test.ts:232]: As noted during pair programming - Using the `http` and `node-fetch` libraries directly is _easier_ (LoC and maintainability). I have concerns about the custom class using in testing that abstracts these

File: test/unit/proxy.test.ts [hunk 2]
```diff
@@ -102,4 +106,107 @@ describe("proxy", () => {
...
+// and specifically tests the proxy in isolation vs. using
+// the httpserver abstraction we've built.
+//
+// Leaving this as a separate test suite for now because
+// we may consider refactoring the httpserver abstraction
+// in the future.
+//
+// If you're writing a test specifically for code in
+// src/node/proxy.ts, you should probably add it to
+// this test suite.
```
Comment (jsjoeio) [inline test/unit/proxy.test.ts:167]: ✍️ @code-asher made a point that the next eng. coming in here might be confused as to where to put their test. I added this comment to address that.

File: test/unit/proxy.test.ts [hunk 3]
```diff
@@ -102,4 +106,107 @@ describe("proxy", () => {
...
+describe("proxy (standalone)", () => {
+  const PORT = 9003
+  const PROXY_PORT = 8003
```
Comment (jsjoeio) [inline test/unit/proxy.test.ts:170]: ❓Chose these ports at random. Not sure if there's a better way to chose two random ports.

Comment (vapurrmaid) [reply test/unit/proxy.test.ts:170]: Could use something like https://www.npmjs.com/package/get-port but it's probably not necessary at this moment in time.

File: test/unit/proxy.test.ts (hunk 2 above)
Comment (vapurrmaid) [reply test/unit/proxy.test.ts:167]: TIL that to obtain a random port, we just create and close an arbitrary server around a locking process, and return the port it was assigned:

https://github.com/sindresorhus/get-port/blob/260c667f457c4260574f915c15c470d2c8eaa12d/index.js#L23-L33

Comment (jsjoeio) [reply test/unit/proxy.test.ts:167]: Nice find! I refactored to use that instead.

File: test/unit/helpers.test.ts [hunk 4]
```diff
@@ -39,3 +39,15 @@ describe("useEnv", () => {
...
+
+describe("getAvailablePort", () => {
+  it("should return a valid port", async () => {
+    const port = await getAvailablePort()
+    expect(port).toBeGreaterThan(0)
+  })
+  it("should return different ports for different calls", async () => {
+    const portOne = await getAvailablePort()
+    const portTwo = await getAvailablePort()
+    expect(portOne).not.toEqual(portTwo)
```
Comment (vapurrmaid) [inline test/unit/helpers.test.ts:51]: I'd add an `expect` that the ports are within the valid port range (0 to 65535)

Comment (jsjoeio) [reply test/unit/helpers.test.ts:51]: Good call! TIL there's a specific range!

**Review by vapurrmaid — APPROVED** [review]

**Review by oxy — APPROVED** [review]
Looks good to me! :sparkles:

**Pull request merged**